src/endless/WMI.cpp

```diff
@@ -553,6 +559,66 @@ HRESULT BcdBootmgr::RemoveFromDisplayOrder(const CString &guid)
     return S_OK;
 }
 
+HRESULT BcdBootmgr::GetTimeout(ULONGLONG & timeout)
+{
+    FUNCTION_ENTER;
+    const int32_t type = BcdBootMgrInteger_Timeout;
+
+    CComPtr<IWbemClassObject> pOutParams;
+    IFFAILED_RETURN_RES(
+        ExecMethod(L"GetElement", &pOutParams,
+            L"Type", type),
+        "Can't get current Timeout");
+
+    CComVariant vTimeoutElement;
+    IFFAILED_RETURN_RES(Get(pOutParams, L"Element", &vTimeoutElement), "Getting Element out param failed");
+
+    CComPtr<IWbemClassObject> pElement = (IWbemClassObject*)vTimeoutElement.byref;
+    CComVariant vTimeout;
+    IFFAILED_RETURN_RES(
+        WmiObject::Get(pElement, L"Integer", &vTimeout),
+        "Getting Integer from Element failed");
+
+    // Although this value is conceptually an uint64, it's passed as a string.
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/aa362650(v=vs.85).aspx
+    wchar_t *wszTimeout = vTimeout.bstrVal;
+    wchar_t *end = NULL;
+    timeout = wcstoull(wszTimeout, &end, 10);
+
+    if (*end != 0) {
+        PRINT_ERROR_MSG_FMT("Failed to parse Timeout value \"%ls\"", wszTimeout);
+        return E_FAIL;
+    }
+
+    return S_OK;
+}
+
+// https://msdn.microsoft.com/en-us/library/windows/desktop/aa362696(v=vs.85).aspx
+// documents the Integer parameter to be of type UI8 but only I4 works in practice.
+HRESULT BcdBootmgr::SetTimeout(const int32_t timeout)
+{
+    FUNCTION_ENTER_FMT("%d", timeout);
```

Comment:
The crash in the previous version of this branch was caused by this `%d` being typoed as `%s`.